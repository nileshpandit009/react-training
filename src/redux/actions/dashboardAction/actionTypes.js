export const DASHBOARD_CONSTANTS = {
  FETCH_POSTS: "FETCH_POSTS",
  FETCH_POSTS_SUCCESS: "FETCH_POSTS_SUCCESS",
  FETCH_POSTS_ERROR: "FETCH_POSTS_ERROR",

  FETCH_POST_DETAILS: "FETCH_POST_DETAILS",
  FETCH_POST_DETAILS_SUCCESS: "FETCH_POST_DETAILS_SUCCESS",
  FETCH_POST_DETAILS_ERROR: "FETCH_POST_DETAILS_ERROR",

  RESET_POSTS: "RESET_POSTS",
  RESET_POST_DETAILS: "RESET_POST_DETAILS",
};
