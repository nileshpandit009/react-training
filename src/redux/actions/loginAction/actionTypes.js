export const LOGIN_CONSTANTS = {
  EMAIL: "EMAIL",
  PASSWORD: "PASSWORD",
  EMAIL_ERR: "EMAIL_ERR",
  PASSWORD_ERR: "PASSWORD_ERR",
  LOGIN_ERR: "LOGIN_ERR",
  RESET_LOGIN: "RESET_LOGIN",
  LOGIN_REQUEST: "LOGIN_REQUEST",
  LOGIN_SUCCESS: "LOGIN_SUCCESS",
  LOGIN_FAILURE: "LOGIN_FAILURE",
};
