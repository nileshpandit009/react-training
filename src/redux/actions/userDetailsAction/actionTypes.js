export const USER_DETAILS_CONSTANTS = {
  USER_DETAILS_REQUEST: "USER_DETAILS_REQUEST",
  USER_DETAILS_SUCCESS: "USER_DETAILS_SUCCESS",
  USER_DETAILS_FAILURE: "USER_DETAILS_FAILURE",
  RESET_USER_DETAILS: "RESET_USER_DETAILS",
};
