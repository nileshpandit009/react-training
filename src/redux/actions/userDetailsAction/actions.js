import { USER_DETAILS_CONSTANTS } from "./actionTypes";

const {
  USER_DETAILS_REQUEST,
  USER_DETAILS_SUCCESS,
  USER_DETAILS_FAILURE,
  RESET_USER_DETAILS,
} = USER_DETAILS_CONSTANTS;

export const fetchUserDetails = (payload) => ({
  type: USER_DETAILS_REQUEST,
  payload,
});

export const userDetailsSuccess = (payload) => ({
  type: USER_DETAILS_SUCCESS,
  payload,
});

export const userDetailsFailure = (payload) => ({
  type: USER_DETAILS_FAILURE,
  payload,
});

export const resetUserDetails = () => ({
  type: RESET_USER_DETAILS,
});
