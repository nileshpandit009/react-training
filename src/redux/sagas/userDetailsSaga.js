import { call, put, takeEvery } from "@redux-saga/core/effects";
import { getUserDetails } from "apis/userDetailsApi";
import {
  userDetailsFailure,
  userDetailsSuccess,
} from "../actions/userDetailsAction/actions";
import { USER_DETAILS_CONSTANTS } from "../actions/userDetailsAction/actionTypes";

function* userDetailsRequest(action) {
  try {
    const response = yield call(getUserDetails);
    yield put(userDetailsSuccess({ userDetails: response.data }));
  } catch (e) {
    yield put(userDetailsFailure({ error: e.message }));
  }
}

export function* userDetailsSaga() {
  yield takeEvery(
    USER_DETAILS_CONSTANTS.USER_DETAILS_REQUEST,
    userDetailsRequest
  );
}
