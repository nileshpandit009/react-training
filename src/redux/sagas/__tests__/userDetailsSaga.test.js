import { put, takeEvery } from "@redux-saga/core/effects";
import {
  userDetailsFailure,
  userDetailsSuccess,
} from "redux/actions/userDetailsAction/actions";
import { USER_DETAILS_CONSTANTS } from "redux/actions/userDetailsAction/actionTypes";
import { userDetailsRequest, userDetailsSaga } from "../userDetailsSaga";

describe("Test user details saga", () => {
  let userDetailsRequestGenerator;
  beforeEach(() => {
    userDetailsRequestGenerator = userDetailsRequest();
    const callDescriptor = userDetailsRequestGenerator.next().value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it("must call userDetailsSuccess action", () => {
    const response = {
      data: {},
    };

    const putDescriptor = userDetailsRequestGenerator.next(response).value;
    expect(putDescriptor).toEqual(
      put(userDetailsSuccess({ userDetails: response.data }))
    );
    expect(userDetailsRequestGenerator.next().done).toBeTruthy();
  });

  it("must call userDetailsFailure action", () => {
    const errResponse = { message: "some error message" };
    const putDescriptor = userDetailsRequestGenerator.throw(errResponse).value;
    expect(putDescriptor).toEqual(
      put(userDetailsFailure({ error: errResponse.message }))
    );

    expect(userDetailsRequestGenerator.next().done).toBeTruthy();
  });
});

describe("Test user details saga watchers", () => {
  let userDetailsSagaItr;
  beforeAll(() => {
    userDetailsSagaItr = userDetailsSaga();
  });

  it("must start task to watch for USER_DETAILS_REQUEST action", () => {
    const takeEveryDescriptor = userDetailsSagaItr.next().value;
    expect(takeEveryDescriptor).toEqual(
      takeEvery(USER_DETAILS_CONSTANTS.USER_DETAILS_REQUEST, userDetailsRequest)
    );
  });
});
