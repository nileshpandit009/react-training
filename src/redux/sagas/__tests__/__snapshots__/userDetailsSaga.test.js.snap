// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test user details saga must call userDetailsFailure action 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`Test user details saga must call userDetailsSuccess action 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
